### DISCUSSION TITLE
gnome-terminal runs whenever I start a GUI app

### DISCUSSION THREAD

**Discussion by jeffguyperaton** [i3/i3#5589] [Q&A] [answered]
Regolith wraps i3. The default config allows me to launch a program with super+space, which works as expected: The firefox window pops up. Or I can launch a program with super+shift+space, which doesn't work as expected. It launches a terminal for stdout and stderr *and* it launches the application window.

This is true whether I launch firefox, chromium, Libreoffice Calc, etc.

I think these are the relevant parts of my config, but I posted the full config at github [click here](https://github.com/jeffguyperaton/i3_config/commit/7783d796aea958d312476daaf24fe7aedf2584da).

#This works correctly: The application window opens, but no terminal window opens.
set_from_resource $i3-wm.binding.launcher.app i3-wm.binding.launcher.app space
set_from_resource $i3-wm.program.launcher.app i3-wm.program.launcher.app ilia -p apps
bindsym $mod+$i3-wm.binding.launcher.app exec --no-startup-id $i3-wm.program.launcher.app -t $ilia.stylesheet

#This works incorrectly: A terminal window opens, containing stdout and stderr, *and* the application window opens.
set_from_resource $i3-wm.binding.launcher.cmd i3-wm.binding.launcher.cmd Shift+space
set_from_resource $i3-wm.program.launcher.cmd i3-wm.program.launcher.cmd ilia -p terminal
bindsym $mod+$i3-wm.binding.launcher.cmd exec --no-startup-id $i3-wm.program.launcher.cmd -t $ilia.stylesheet

Thanks for any help you can provide!

**Comment by okraits** [comment]
It's hard or nearly impossible to find the issue without having a look at your i3 configuration. Please provide it.
Maybe it's also specific to your distribution and the setup you're using.

**Reply by jeffguyperaton** [reply]
Thanks! I couldn't figure out how to attach a tarball of my config file, so I uploaded it to github and updated the original post with a link. Is there a better way for me to have included my config file?

Thanks!

**Comment by jeffguyperaton** [comment, marked as answer]
Any time I want to use a new app, create a new file following the model of /usr/share/applications/firefox.desktop, then use super+space to run that app. Problem solved!

**Reply by orestisfl** [reply]
One usually doesn't need to create the `.desktop` files as your distro should provide those for regular graphical apps. If you do create your own you can place them under `~/.local/share/applications/`.

Looking at [ilia's](https://github.com/regolith-linux/ilia) readme, it says:

> System Commands (terminal)
> 
> Loads all system commands available on the path. Selecting a command will cause it to be executed from a newly created terminal.

So, what you are seeing is correct for your shift+space binding, it is supposed to open a terminal.
